### PR TITLE
Add helper to test for potential mesh errors

### PR DESCRIPTION
**Pull request by julz**
# Changes

- :gift: Add helper to test for potential mesh errors. 
   - This is useful to distinguish between errors which should and shouldnt cause fallback to clusterIP (see https://github.com/knative/serving/pull/11208, https://github.com/knative/serving/pull/11174)

/kind enhancement

**Release Note**

```release-note
Adds `IsPotentialMeshErrorResponse` helper to aid distinguishing between responses that should or should not cause fall-back to cluster IP.
```

/assign @markusthoemmes @vagababov

### CODE REVIEW

File: pkg/network_test.go [hunk 1]
```diff
@@ -675,3 +676,31 @@ func TestPortNumberForName(t *testing.T) {
...
+		statusCode int
+		expect     bool
+	}{{
+		statusCode: 404,
+		expect:     false,
+	}, {
+		statusCode: 200,
+		expect:     false,
+	}, {
+		statusCode: 200,
```
Comment (nak3) [inline pkg/network_test.go:691]: This is duplicated with above `statusCode: 200` test?

Comment (julz) [reply pkg/network_test.go]: Heh, thanks for the catch @nak3, fixed! 🤦‍♂️

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**